packages/component-base/src/dom-utils.js

```diff
@@ -40,6 +40,27 @@ export function getAncestorRootNodes(node) {
   return result;
 }
 
+/**
+ * Returns the list of flattened elements for the given `node`.
+ * This list consists of a node's children and, for any children that are
+ * `<slot>` elements, the expanded flattened list of `assignedElements`.
+ *
+ * @param {Node} node
+ * @return {Element[]}
+ */
+export function getFlattenedElements(node) {
+  const result = [];
+  let elements;
+  if (node.localName === 'slot') {
+    elements = node.assignedElements({ flatten: true });
```

Comment:
I'm a bit concerned that a node inside a nested slot might end up appearing twice in the resulting array, given that we do `flatten: true` here.

Reply:
Thanks, makes sense. Removed.